src/components/ListDetail/ListDetail.tsx

```diff
@@ -0,0 +1,299 @@
+    const next =
+      index === listDetailItemRefs.length - 1 ? 0 : index + 1; /* 2 */
+    const prev =
+      index === 0 ? listDetailItemRefs.length - 1 : index - 1; /* 2 */
+
+    if ([R_ARROW_KEYCODE, D_ARROW_KEYCODE].includes(e.code)) {
+      /* 3 */
+      listDetailItemRefs[next].current.focus();
+    } else if ([L_ARROW_KEYCODE, U_ARROW_KEYCODE].includes(e.code)) {
+      /* 4 */
+      listDetailItemRefs[prev].current.focus();
+    }
+  }
+
+  const childrenWithProps = React.Children.map(
+    listDetailItems(),
+    (child: any, i: any) => {
+      // Checking isValidElement is the safe way and avoids a typescript
+      // error too.
+      if (React.isValidElement(child)) {
+        return React.cloneElement<Props>(child, {
+          id: idVar[i],
+          ariaLabelledBy: ariaLabelledByVar[i],
+        });
+      }
+      return child;
+    },
+  );
+  const TagName = variant === 'ordered' ? 'ol' : 'ul';
+  const componentClassName = clsx(styles['list-detail'], className, {});
+
+  return (
+    <div className={componentClassName} {...other}>
+      <div className={styles['list-detail__header']}>
+        <TagName
+          className={clsx(styles['list-detail__list'], {
+            [styles['list-detail__list--ordered']]: variant === 'ordered',
+          })}
+          role="tablist"
+        >
+          {listDetailItems().map((tab: any, i: any) => {
+            const isActive = activeIndexState === i;
+            const itemVariant = variant && tab.props.variant;
+            return (
+              <li
+                className={clsx(styles['list-detail__item'], {
+                  [styles['eds-is-active']]: isActive,
+                  [styles['list-detail__item--success']]:
+                    itemVariant === 'success',
+                  [styles['list-detail__item--warning']]:
+                    itemVariant === 'warning',
+                  [styles['list-detail__item--error']]: itemVariant === 'error',
+                })}
+                key={'list-detail-item-' + i}
+                role="presentation"
+              >
+                <a
+                  className={styles['list-detail__link']}
+                  role="tab"
+                  onClick={(e) => {
+                    e.preventDefault();
+                    onOpen(i);
+                  }}
+                  key={'list-detail-link' + i}
+                  id={ariaLabelledByVar[i]}
+                  onKeyDown={onKeyDown}
+                  href={`#${idVar[i]}`}
+                  aria-selected={isActive}
+                  tabIndex={isActive ? 0 : -1}
+                  aria-controls={idVar[i]}
+                  ref={listDetailItemRefs[i]}
+                  aria-label={tab.props.ariaLabel}
+                >
+                  <div className={clsx(styles['list-detail__link-left'], {
+                    [styles['list-detail__link-hidden']]: itemVariant === 'number'
+                  })}>
+                    {itemVariant === 'success' ? (
```

Comment:
I started to abstract it into a function but it looked even messier. It would take 2 args (itemVariant and i) and return either an Icon or a span... I didn't see it looking any tidier than this.